### PR TITLE
Allow custom form for batch actions

### PR DESCRIPTION
**Pull request by giaptx**
Hi @jowilf,

I would like to create a PR to try to resolve this issue https://github.com/jowilf/starlette-admin/issues/21#issuecomment-1354299135

I'm not master on frontend, but I try to make this works for my application. The idea is each action will have own form html to custom the input for each action.

Could you have a look? Please feel free to reject if you have any better plan or solution.

Thanks

### CODE REVIEW

File: starlette_admin/actions.py [hunk 1]
```diff
@@ -7,6 +7,7 @@ def action(
     confirmation: Optional[str] = None,
     submit_btn_text: Optional[str] = "Proceed",
     submit_btn_class: Optional[str] = "btn-primary",
+    form: str = "",
```
Comment (jowilf) [inline starlette_admin/actions.py:10]:
```suggestion
    form: Optional[str] = None,
```

File: starlette_admin/statics/js/list.js [hunk 2]
```diff
@@ -423,6 +423,11 @@ $(function () {
       query.append("pks", s);
     });
     query.append("name", name);
+
+    $('#modal-action :input').each(function() {
+        query.append(this.name, $(this).val());
+    });
+
```
Comment (jowilf) [inline starlette_admin/statics/js/list.js:430]: Send form data as [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData)

Comment (jowilf) [reply starlette_admin/statics/js/list.js:430]: Useful link -> https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects#retrieving_a_formdata_object_from_an_html_form

File: examples/custom_actions/models.py [hunk 3]
```diff
@@ -36,14 +36,20 @@ class ArticleView(ModelView):
         confirmation="Are you sure you want to mark selected articles as published ?",
         submit_btn_text="Yes, proceed",
         submit_btn_class="btn-success",
+        form="""<div class="form-floating mb-3">
```
Comment (jowilf) [inline examples/custom_actions/models.py:39]: I'm not sure but i think a template path could be better here, wdyt?

**Review by jowilf — CHANGES_REQUESTED** [review]
Hello @giaptx, 
It's better idea, we should find a better and easy way to include that,
I made some few comments, please check them

Please add some test, update docstring and the documentation

**Comment by giaptx** [comment]
Thanks for your review. I will improve when I get time.
Feel free to push to PR if you get time too, because you are best.

**Review by jowilf — APPROVED** [review]

**Comment by jowilf** [comment]
Thank you all for your contributions

**Pull request merged**